velox/exec/HashPartitionFunction.h

```diff
@@ -58,4 +58,36 @@ class HashPartitionFunction : public core::PartitionFunction {
   SelectivityVector rows_;
   raw_vector<uint64_t> hashes_;
 };
+
+/// Factory class to create HashPartitionFunction
+/// 'keyChannels_' stores the index of keys to partition on, if the key is a
+/// constant, use index 'kConstantChannel' to indicate so and store the constant
+/// value as a base vector in 'constValues_'
+/// The 'constValues_' size is less than or equal to 'keyChannels_' size
+class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
+ public:
+  HashPartitionFunctionSpec(
+      RowTypePtr inputType,
+      std::vector<column_index_t> keyChannels,
+      std::vector<VectorPtr> constValues = {})
```

Comment:
@kewang1024 Do we need this default value? Let's remove it.

Reply:
I also thought about this one, but I'm more leaning towards keeping it given:
1. This call is used in several places, and having the default value makes the code cleaner and more straighforward
2. From the default value, it's easy to infer that constValues_.size != keyChannels_.size, it's a good implications for the relationships between constValues_ and keyChannels_

Let me know if you have strong opinion against it